src/main/java/org/ice4j/socket/MultiplexingDatagramSocket.java

```diff
@@ -138,6 +148,25 @@ public int getReceiveBufferSize()
     public MultiplexingDatagramSocket()
         throws SocketException
     {
+        this(false);
+    }
+
+    /**
+     * Initializes a new <tt>MultiplexingDatagramSocket</tt> instance which is
+     * to enable <tt>DatagramPacket</tt> filtering and binds it to any available
+     * port on the local host machine. The socket will be bound to the wildcard
+     * address, an IP address chosen by the kernel.
+     *
+     * @throws SocketException if the socket could not be opened, or the socket
+     * could not bind to the specified local port
+     * @param persistent whether this socket should be kept open after all of
+     * its {@link MultiplexedDatagramSocket}s are closed.
+     * @see DatagramSocket#DatagramSocket()
+     */
+    public MultiplexingDatagramSocket(boolean persistent)
+        throws SocketException
+    {
+        this.persistent = persistent;
```

Comment:
ok now i see there are all sorts of ctors here :/ i was hoping we could keep the main ctor logic to one of them but doesn't look like this would help much.